contracts/carrot-app/src/handlers/query.rs

```diff
@@ -79,94 +85,99 @@ fn query_compound_status(deps: Deps, env: Env, app: &App) -> AppResult<CompoundS
     })
 }
 
-fn query_position(deps: Deps) -> AppResult<PositionResponse> {
-    let position = POSITION.may_load(deps.storage)?;
+pub fn query_strategy(deps: Deps) -> AppResult<StrategyResponse> {
+    let config = CONFIG.load(deps.storage)?;
+
+    Ok(StrategyResponse {
+        strategy: config.balance_strategy.into(),
+    })
+}
 
-    Ok(PositionResponse { position })
+pub fn query_strategy_status(deps: Deps, app: &App) -> AppResult<StrategyResponse> {
+    let config = CONFIG.load(deps.storage)?;
+
+    Ok(StrategyResponse {
+        strategy: config
+            .balance_strategy
+            .query_current_status(deps, app)?
+            .into(),
+    })
 }
 
 fn query_config(deps: Deps) -> AppResult<Config> {
     Ok(CONFIG.load(deps.storage)?)
 }
-fn query_balance(deps: Deps, _app: &App) -> AppResult<AssetsBalanceResponse> {
-    let pool = get_osmosis_position(deps)?;
 
-    let balances = try_proto_to_cosmwasm_coins(vec![pool.asset0.unwrap(), pool.asset1.unwrap()])?;
-    let liquidity = pool.position.unwrap().liquidity.replace('.', "");
+pub fn query_balance(deps: Deps, app: &App) -> AppResult<AssetsBalanceResponse> {
```

Comment:
Would be nice to add pagination here

Reply:
Are we there yet @CyberHoward, especially the balance query would be misleading with pagination (but could still run out of gas)